bgpd/bgp_evpn_vty.c

```diff
@@ -4120,12 +4124,20 @@ DEFUN(show_bgp_l2vpn_evpn_route,
 		/* Specific type is requested */
 		if (strncmp(argv[type_idx + 1]->arg, "ma", 2) == 0)
 			type = BGP_EVPN_MAC_IP_ROUTE;
+		else if (strncmp(argv[type_idx + 1]->arg, "2", 1) == 0)
```

Comment:
same comment i.e. instead of checking for one number in each strcmp just convert it to an integer and if that conversion fails use as string.

Reply:
you still need to know the number so might as well do a strcmp here, @taspelund you can just make it
```c
if (strmatch(argv[type_idx + 1]->text, "macip") || strmatch(argv[type_idx + 1], "2"))
   type = BGP_EVPN_MACIP_ROUTE;
```

if you switched to defpy:
```c
if (strmatch(type_name, "macip") || strmatch(type_name, "2"))
   type = BGP_EVPN_MACIP_ROUTE;
```